services/datamanager/builtin/capture/capture.go

```diff
@@ -52,11 +58,15 @@ type Capture struct {
 
 	collectorsMu sync.Mutex
 	collectors   collectors
-
 	// captureDir is only stored on Capture so that we can detect when it changs
 	captureDir string
 	// maxCaptureFileSize is only stored on Capture so that we can detect when it changs
 	maxCaptureFileSize int64
+
+	mongoMU         sync.Mutex
+	mongoClient     *mongo.Client
+	mongoCollection *mongo.Collection
```

Comment:
Why are we keeping a pointer to the Mongo collection _and_ client? Is it to close the connection at shutdown?

Reply:
1. Because the client has the `Disconnect` method, which we need to call on close / if the user removes the mongo_uri from the config
2. The following scenario illustrates why we need to remember the mongoCollection pointer:

```
1. User configures a valid mongo URL with a single collector (a sensor). Data capture creates a mongoClient which succeeds, then creates a mongoCollection. It creates a collector and passes it to the new collector.  
2. The user configures a new collector for a movement-sensor. Data capture should create the collector & give the collector the same mongoCollection pointer, which we can't do if we don't remember the pointer we created in step 1.
```

I didn't want to create a new `*mongo.Collection` pointer every time we create a new collector as I don't know the potential performance impact of doing so.

That said, you know the mongo client API better than I do. Do you see a better way to do what I'm trying to do?

Reply:
Makes sense. No I think this is the best way. I sort of answered this question myself as I was reviewing, probably should have just removed it :)